src/paymentRequest/utils.ts

```diff
@@ -1,16 +1,71 @@
-import { hexToBuffer } from '@celo/utils/lib/address'
 import { decryptComment, encryptComment } from '@celo/cryptographic-utils'
+import { isE164NumberStrict } from '@celo/phone-utils'
+import { hexToBuffer } from '@celo/utils/lib/address'
+import BigNumber from 'bignumber.js'
 import { call } from 'redux-saga/effects'
 import { MAX_COMMENT_LENGTH } from 'src/config'
 import { features } from 'src/flags'
 import i18n from 'src/i18n'
 import { PaymentRequest, WriteablePaymentRequest } from 'src/paymentRequest/types'
+import { Recipient } from 'src/recipients/recipient'
+import { TransactionDataInput } from 'src/send/SendAmount'
+import { TokenBalance } from 'src/tokens/slice'
 import Logger from 'src/utils/Logger'
+import { Currency } from 'src/utils/currencies'
 import { doFetchDataEncryptionKey } from 'src/web3/dataEncryptionKey'
-import { isE164NumberStrict } from '@celo/phone-utils'
 
 const TAG = 'paymentRequest/utils'
 
+// Convert a payment request into a transaction data object if user has enough balance
+export const transactionDataFromPaymentRequest = ({
+  paymentRequest,
+  stableTokens,
+  requester,
+}: {
+  paymentRequest: PaymentRequest
+  stableTokens: TokenBalance[]
+  requester: Recipient
+}): TransactionDataInput | undefined => {
```

Comment:
might be cleaner to just throw an error instead of returning "undefined" when there's insufficient balance